### PR TITLE
chore(edit-ema) #27301  Add functionality for empty containers  

### LINKED ISSUE #27301: Edit EMA: Add functionality for empty containers
### Parent Issue

https://github.com/dotCMS/core/issues/26644

### Task

Right now we don't have an element for empty containers therefore if we add new containers or delete all the contentlet from a container, it just disappears and we cannot add more contentlets to it

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

* The client should be able to add contentlets to an empty container 
* The client should be able to fill a container after deleting all the contentlets
* We are missing the design, but for now is enough to make it work.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

image

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed changes

- Add interactive content when a container is empty.

- This empty container have the Edit EMA Tools, but only the option to add new Content | Widget | Form

https://github.com/dotCMS/core/assets/144152756/0c0db045-8760-4ee7-9627-309e818976cb

 
### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/sdk/react/src/lib/components/Container/Container.spec.tsx [hunk 1]
```diff
@@ -31,6 +31,8 @@ describe('Container', () => {
         expect(screen.getByTestId('no-component')).toHaveTextContent(
             'No Component for content-type-1'
         );
+
+        expect(screen.getByTestId('empty-container')).toHaveTextContent('This container is empty.');
```
Comment (fmontes) [inline core-web/libs/sdk/react/src/lib/components/Container/Container.spec.tsx:35]: Can we have both cases in the same container?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 2]
```diff
@@ -154,4 +154,15 @@ export class EmaContentletToolsComponent {
...
     }
+
+    /**
+     *
+     *
+     * @readonly
+     * @type {boolean}
+     * @memberof EmaContentletToolsComponent
+     */
+    get isTempContentletEmpty(): boolean {
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:165]:
```suggestion
    get isContainerEmpty(): boolean {
```

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 3]
```diff
@@ -154,4 +154,15 @@ export class EmaContentletToolsComponent {
             zIndex: '1'
         };
     }
+
+    /**
+     *
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:159]: Docs needs description... Github Copilot Chat can help.

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx [hunk 4]
```diff
@@ -6,6 +6,36 @@ import { PageContext } from '../../contexts/PageContext';
...
+    identifier: 'TEMP_EMPTY_CONTENTLET',
+    title: 'TEMP_EMPTY_CONTENTLET',
+    contentType: 'TEMP_EMPTY_CONTENTLET_TYPE',
+    inode: 'TEMPY_EMPTY_CONTENTLET_INODE',
+    widgetTitle: 'TEMP_EMPTY_CONTENTLET'
+};
+
+function getEmptyContent(contentType: string) {
+    return contentType === 'TEMP_EMPTY_CONTENTLET_TYPE' ? EmptyContainer : NoContent;
+}
```
Comment (fmontes) [inline core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:19]: This is not a good approach, we should not mix the concepts of "empty container" and "there is no component for this content type".

Just render directly your fake container when `contentlet.lenght` === 0.

Make sure is very clear the intention of this here.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/react/src/lib/components/Container/Container.spec.tsx (hunk 1 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/react/src/lib/components/Container/Container.spec.tsx:35]: No, i added other test for this case

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx (hunk 4 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:19]: Solved!

**Review by fmontes — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

**Pull request merged**